src/Mbstring/Mbstring.php

```diff
@@ -533,18 +535,20 @@ public static function mb_str_split($string, $split_length = 1, $encoding = null
 
             return null;
         }
-
-        if (1 > $split_length = (int) $split_length) {
-            trigger_error('The length of each segment must be greater than zero', E_USER_WARNING);
-
-            return false;
+        if (null !== $split_length && !\is_bool($split_length) && !\is_numeric($split_length)) {
+            trigger_error('mb_str_split() expects parameter 2 to be int, '.\gettype($split_length).' given', E_USER_WARNING);
+            
+            return null;
         }
+        $split_length = (int) $split_length;
+        if (1 > $split_length) {
+            trigger_error('mb_str_split(): The length of each segment must be greater than zero', E_USER_WARNING);
 
-        if (null === $encoding) {
-            $encoding = mb_internal_encoding();
+            return false;
         }
 
-        if ('UTF-8' === $encoding = self::getEncoding($encoding)) {
+        $encoding = self::getEncoding($encoding);
+        if ('UTF-8' === $encoding) {
```

Comment:
this change is not needed

Reply:
@nicolas-grekas more readable

Reply:
unrelated and subjective, complicates the review

Reply:
Whether this is more readable is probably debatable, but that question is out of scope for this PR.

If the actual change is hidden among CS fixes, a PR is really hard to review.